selfservice/strategy/oidc/provider_lark.go

```diff
@@ -0,0 +1,112 @@
+// Copyright © 2022 Ory Corp
+// SPDX-License-Identifier: Apache-2.0
+
+package oidc
+
+import (
+	"context"
+	"encoding/json"
+	"net/url"
+
+	"github.com/hashicorp/go-retryablehttp"
+	"github.com/pkg/errors"
+	"golang.org/x/oauth2"
+
+	"github.com/ory/herodot"
+	"github.com/ory/x/httpx"
+)
+
+type ProviderLark struct {
+	*ProviderGenericOIDC
+}
+
+func NewProviderLark(
+	config *Configuration,
+	reg dependencies,
+) *ProviderLark {
+	return &ProviderLark{
+		&ProviderGenericOIDC{
+			config: config,
+			reg:    reg,
+		},
+	}
+}
+
+func (g *ProviderLark) Config() *Configuration {
+	return g.config
+}
+
+func (g *ProviderLark) OAuth2(ctx context.Context) (*oauth2.Config, error) {
+	var endpoint = oauth2.Endpoint{
+		AuthURL:   "https://passport.feishu.cn/suite/passport/oauth/authorize",
+		TokenURL:  "https://passport.feishu.cn/suite/passport/oauth/token",
+		AuthStyle: oauth2.AuthStyleInParams,
+	}
+
+	return &oauth2.Config{
+		ClientID:     g.config.ClientID,
+		ClientSecret: g.config.ClientSecret,
+		Endpoint:     endpoint,
+		// Lark uses fixed scope that can not be configured in runtime
+		Scopes:      g.config.Scope,
+		RedirectURL: g.config.Redir(g.reg.Config().OIDCRedirectURIBase(ctx)),
+	}, nil
+
+}
+
+func (g *ProviderLark) Claims(ctx context.Context, exchange *oauth2.Token, query url.Values) (*Claims, error) {
+	// larkClaim is defined in the https://open.feishu.cn/document/common-capabilities/sso/api/get-user-info
+	type larkClaim struct {
+		Sub          string `json:"sub"`
+		Name         string `json:"name"`
+		Picture      string `json:"picture"`
+		OpenID       string `json:"open_id"`
+		UnionID      string `json:"union_id"`
+		EnName       string `json:"en_name"`
+		TenantKey    string `json:"tenant_key"`
+		AvatarURL    string `json:"avatar_url"`
+		AvatarThumb  string `json:"avatar_thumb"`
+		AvatarMiddle string `json:"avatar_middle"`
+		AvatarBig    string `json:"avatar_big"`
+		Email        string `json:"email"`
+		UserID       string `json:"user_id"`
+		Mobile       string `json:"mobile"`
+	}
+	var (
+		userEndpoint = "https://passport.feishu.cn/suite/passport/oauth/userinfo"
+		accessToken  = exchange.AccessToken
+		client       = g.reg.HTTPClient(ctx, httpx.ResilientClientDisallowInternalIPs())
+		user         larkClaim
+	)
+
+	req, err := retryablehttp.NewRequest("GET", userEndpoint, nil)
+	if err != nil {
+		return nil, errors.WithStack(herodot.ErrInternalServerError.WithReasonf("%s", err))
+	}
+
+	req.Header.Add("Authorization", "Bearer "+accessToken)
```

Comment:
do we need to manually add Authorization Bearer header here? 
If we use client like this
```
client := g.reg.HTTPClient(ctx, httpx.ResilientClientDisallowInternalIPs(), httpx.ResilientClientWithClient(o.Client(ctx, exchange)))
```
Then the client will already use provided token for the request (ensured in `o.Client(ctx, exchange)`)

Reply:
I appreciate your suggestion.

This [line](https://github.com/ory/x/blob/d5dfdaaee6a9432c4889c1224b705d8ccfe3219d/httpx/resilient_client.go#L124) will override the Transport settings of HTTP client and the token will not be sent if the user uses tracing.

So I prefer manually set the header here.  A better way is to use `SetAuthHeader` method.

Reply:
Hey @CNLHC  sorry for late response :) You can use the client like I suggested, the issue with transport being overridden was not in otel and set tracer, the logic there was correctly applying transport configuration. Issue was in NoInternalIPRoundTripper, it was ignoring already set transport :) 
Please rebase your branch with master :) Fix is already there.

Reply:
Thanks for your comment and I will rebase this branch to the master ASAP.